### ISSUE TITLE
Introduce a process manager

### ISSUE THREAD

**Issue by thomaseizinger** [itchysats/itchysats#956]
The end-goal of the process manager is to de-couple components from each other and perform the "if this event happens, then I want that component to do X". For example, "if contract setup completes, we need to start monitoring".

The process manager should be an actor with a handler for a _single_ event. Only an event that is successfully saved to the DB must be processed further. If saving an event fails (for example, due to [OCC](https://en.wikipedia.org/wiki/Optimistic_concurrency_control)), we have to report that **back** to the caller to inform them that whatever command they ran operated on outdated data.

The process manager should be decoupled an not know about the specific other actors. It thus cannot use addresses but SHOULD use one`MessageChannel`s per action that it sets off. For example `MessageChannel<StartMonitoring>`.

In a nutshell, the flow within `ProcessManager` will be:

1. Handler with new event gets called
2. Save it to DB
3. `match` on the event
4. Decide based on the event, which message to send into which `MessageChannel`

Initially, we can skip implementing OCC.
